src/audio/base_fw.c

```diff
@@ -431,6 +433,48 @@ static int basefw_power_state_info_get(uint32_t *data_offset, char *data)
 	return 0;
 }
 
+static int basefw_libraries_info_get(uint32_t *data_offset, char *data)
+{
+	struct ipc4_libraries_info *const libs_info = (struct ipc4_libraries_info *)data;
+	struct sof_man_fw_desc *desc;
+
+	libs_info->library_count = 0;
+
+	for (int lib_id = 0; lib_id < LIB_MANAGER_MAX_LIBS; ++lib_id) {
+#ifdef ZEPHYR_SOC_INTEL_ADSP_MEMORY_H_
```

Comment:
This condition is surprising. What is it supposed to check, apart from a header file is included?

Reply:
This is exactly what he is supposed to check. Generally it is that in uses the definition which is platform specific. The idea is to check if there is this definition which is in the code. Do you think it's better to replace it with CONFIG_INTEL?

Reply:
@kv2019i and I were discussing a ```src/ipc/ipc4/intel.c``` where we could put a lot of the vendor specific code and wrap it with Kconfigs, have it read values from Zephyr DT later too.

It would seem we could use that here too and rather than using the ```#ifdef ZEPHYR_SOC_INTEL_ADSP_MEMORY_H_``` we use a Kconfig ```CONFIG_SOME_INTEL_MEMORY_OPTION```, so lets use Kconfig now and next step move into an intel specific file.

Reply:
My question arises from the fact that in another place in the code where the `IMR_BOOT_LDR_MANIFEST_BASE` definition is used, the condition looks different. It would be a good to check it the same way in both places.

https://github.com/thesofproject/sof/blob/14c4e867573096a00634e46975479591ec0257a9/src/ipc/ipc4/helper.c#L959-L961